server/src/main/java/org/spine3/server/entity/IdSetFunctionMap.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2016, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.entity;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Maps;
+import com.google.protobuf.Message;
+import org.spine3.Internal;
+import org.spine3.base.EventContext;
+import org.spine3.server.type.EventClass;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Helper class for managing {@link IdSetEventFunction}s associated with
+ * a repository that dispatches events to its entities.
+ *
+ * @param <I> the type of the entity IDs of this repository
+ * @author Alexander Yevsyukov
+ */
+@Internal
+public class IdSetFunctionMap<I> {
```

Comment:
As long it is not a `Map` per se, I would roll back to the previous name it had, i.e. `IdSetFunctions`.